examples/shader/animate_shader.rs

```diff
@@ -0,0 +1,128 @@
+        .add_system(animate_shader.system())
+        .run();
+}
+
+#[derive(RenderResources, Default, TypeUuid)]
+#[uuid = "463e4b8a-d555-4fc2-ba9f-4c880063ba92"]
+struct TimeComponent {
+    value: f32,
+}
+
+const VERTEX_SHADER: &str = r#"
+#version 450
+
+layout(location = 0) in vec3 Vertex_Position;
+layout(location = 1) in vec2 Vertex_Uv;
+layout(location = 0) out vec2 v_Uv;
+
+layout(set = 0, binding = 0) uniform CameraViewProj {
+    mat4 ViewProj;
+};
+
+layout(set = 1, binding = 0) uniform Transform {
+    mat4 Model;
+};
+
+void main() {
+    gl_Position = ViewProj * Model * vec4(Vertex_Position, 1.0);
+    v_Uv = Vertex_Uv;
+}
+"#;
+
+const FRAGMENT_SHADER: &str = r#"
+#version 450
+
+layout(location = 0) in vec2 v_Uv;
+layout(location = 0) out vec4 o_Target;
+
+layout(set = 2, binding = 0) uniform TimeComponent_value {
+    float u_time;
+};
+
+void main() {
+    float speed = 0.7;
+    float translation = sin(u_time * speed);
+    float percentage = 0.6;
+    float threshold = v_Uv.x + translation * percentage;
+
+    vec3 red = vec3(1., 0., 0.);
+    vec3 blue = vec3(0., 0., 1.);
+    vec3 mixed = mix(red, blue, threshold);
+
+    o_Target = vec4(mixed, 1.0);
+}
+"#;
+
+fn setup(
+    mut commands: Commands,
+    mut pipelines: ResMut<Assets<PipelineDescriptor>>,
+    mut shaders: ResMut<Assets<Shader>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut render_graph: ResMut<RenderGraph>,
+) {
+    // Create a new shader pipeline
+    let pipeline_handle = pipelines.add(PipelineDescriptor::default_config(ShaderStages {
+        vertex: shaders.add(Shader::from_glsl(ShaderStage::Vertex, VERTEX_SHADER)),
+        fragment: Some(shaders.add(Shader::from_glsl(ShaderStage::Fragment, FRAGMENT_SHADER))),
+    }));
+
+    // Add a `RenderResourcesNode` to our `RenderGraph`. This will bind `TimeComponent` to our shader
+    render_graph.add_system_node(
+        "time_component",
+        RenderResourcesNode::<TimeComponent>::new(true),
+    );
+
+    // Add a `RenderGraph` edge connecting our new "time_component" node to the main pass node. This
+    // ensures that "time_component" runs before the main pass
+    render_graph
+        .add_node_edge("time_component", base::node::MAIN_PASS)
+        .unwrap();
+
+    // Spawn a quad and insert the `TimeComponent`
+    commands
+        .spawn_bundle(MeshBundle {
+            mesh: meshes.add(Mesh::from(shape::Quad {
+                size: Vec2::new(5.0, 5.0),
+                flip: true,
+            })),
+            render_pipelines: RenderPipelines::from_pipelines(vec![RenderPipeline::new(
+                pipeline_handle,
+            )]),
+            transform: Transform::from_xyz(0.0, 0.0, 0.0),
+            ..Default::default()
+        })
+        .insert(TimeComponent { value: 0.0 });
+
+    // Spawn a camera
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_xyz(0.0, 0.0, -8.0).looking_at(Vec3::ZERO, -Vec3::Y),
```

Comment:
Regarding: `-Vec3::Y`, without the minus, the colours are flipped. 

I was expecting to have red on the left and blue on right, but it's the opposite. The same happened to me when just drawing the UVs in the fragment shader directly ( `gl_FragColor = vec4(v_Uv, 0.0, 1.0);` ). You get black ( 0,0 coords ) on the top right corner instead of bottom left. 

Where in my code in this example i should fix it, if not via this `-Vec3::Y`? Thanks a lot

Reply:
Using a 2d camera fixed it (and i increase the Quad size).

With `.looking_at(Vec3::ZERO, Vec3::Y)` and a 3d camera it's definitely upside down. `flip: true` / `false` on the Quad doesn't fix it. Is it worth an issue or is there another bug in the example?

In my opinion the current example is correct, but please let me have your feedback.

Reply:
In the last commit i have reverted it to using a 3D camera, with `Transform::from_xyz(0.0, 0.0, 8.0).looking_at(Vec3::ZERO, Vec3::Y)`. So positive Z and `Quad::new` without flipping works.

In my opinion this should be clear enough for users.

Reply:
https://github.com/bevy-cheatbook/bevy-cheatbook/issues/55